src/lib/model/AccountsModel.js

```diff
@@ -0,0 +1,377 @@
+                requests.push(request);
+            }
+            request.partyList.push({
+                partyIdType: account.idType,
+                partyIdentifier: account.idValue,
+                fspId: this.dfspId,
+            });
+        }
+        return requests;
+    }
+
+    /**
+     * Returns an object representing the final state of the transfer suitable for the outbound API
+     *
+     * @returns {object} - Response representing the result of the transfer process
+     */
+    getResponse() {
+        // we want to project some of our internal state into a more useful
+        // representation to return to the SDK API consumer
+        const resp = { ...this.data };
+
+        switch(this.data.currentState) {
+            case 'succeeded':
+                resp.currentState = stateEnum.COMPLETED;
+                break;
+
+            case 'errored':
+                resp.currentState = stateEnum.ERROR_OCCURED;
+                break;
+
+            default:
+                this.logger.log(
+                    `Account model response being returned from an unexpected state: ${this.data.currentState}. ` +
+                    'Returning ERROR_OCCURED state'
+                );
+                resp.currentState = stateEnum.ERROR_OCCURED;
+                break;
+        }
+
+        return resp;
+    }
+
+
+    /**
+     * Persists the model state to cache for reinitialisation at a later point
+     */
+    async _save() {
+        try {
+            this.data.currentState = this.stateMachine.state;
+            const res = await this.cache.set(`accountModel_${this.data.modelId}`, this.data);
+            this.logger.push({ res }).log('Persisted account model in cache');
+        }
+        catch(err) {
+            this.logger.push({ err }).log('Error saving account model');
+            throw err;
+        }
+    }
+
+
+    /**
+     * Loads an accounts model from cache for resumption of the accounts management process
+     *
+     * @param modelId {string} - UUID of the model to load from cache
+     */
+    async load(modelId) {
+        try {
+            const data = await this.cache.get(`accountModel_${modelId}`);
+            await this.initialize(data);
+            this.logger.push({ cache: this.data }).log('Account model loaded from cached state');
+        }
+        catch(err) {
+            this.logger.push({ err }).log('Error loading account model');
+            throw err;
+        }
+    }
+
+
+    /**
+     * Unsubscribes the models subscriber from all subscriptions
+     */
+    async _unsubscribeAll() {
+        return new Promise((resolve) => {
+            this.subscriber.unsubscribe(() => {
+                this.logger.log('Account model unsubscribed from all subscriptions');
+                return resolve();
+            });
+        });
+    }
+
+
+    /**
+     * Returns a promise that resolves when the state machine has reached a terminal state
+     */
+    async run() {
```

Comment:
Please check recent changes in the outbound model - I made a commit few days ago with updates for passing errors back to the backend.

Reply:
I've added necessary changes to the `catch` block.